src/main/java/kristar/projects/services/impl/UserServiceImpl.java

```diff
@@ -47,4 +51,16 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
         User savedUser = userRepository.save(userFromRequestDto);
         return userMapper.toUserResponse(savedUser);
     }
+
+    @Override
+    public User getCurrentUser() throws AccessDeniedException {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication == null || !authentication.isAuthenticated()) {
+            throw new AccessDeniedException("You should be authorized");
+        }
+        String userEmail = authentication.getName();
+
+        return userRepository.findByEmail(authentication.getName()).orElseThrow(()
+                -> new UsernameNotFoundException("User not found by email" + userEmail));
```

Comment:
Do you really need this query to the DB for each request? Is it enough to validate token?